installer/manifests/keptn/templates/_helpers.tpl

```diff
@@ -463,3 +463,66 @@ Usage:
   {{- end }}
 {{- end -}}
 
+{{- define "keptn.initContainers.wait-for-nats" -}}
+- name: "wait-for-nats"
+  image: "{{ .Values.global.initContainers.image }}:{{ .Values.global.initContainers.tag }}"
+  env:
+  - name: "ENDPOINT"
+    value: "{{ .Values.nats.nameOverride }}"
+  command: ['sh', '-c', 'until nslookup $ENDPOINT; do echo waiting for $ENDPOINT; sleep 2; done;']
+{{- end -}}
+
+{{- define "keptn.initContainers.wait-for-keptn-mongo" -}}
+- name: "wait-for-keptn-mongo"
+  image: "{{ .Values.global.initContainers.image }}:{{ .Values.global.initContainers.tag }}"
+  env:
+  - name: "ENDPOINT"
+    value: "keptn-mongo"
+  command: ['sh', '-c', 'until nslookup $ENDPOINT; do echo waiting for $ENDPOINT; sleep 2; done;']
```

Comment:
here we might have a problem if a user has an external mongo installation. I would suggest to do the lookup only if there is no external connection set via `mongo.external.connectionString`

Reply:
this is fixed in the templates for shippy and mongodb-datastore. the waiting is only done if no external connection string is provided in the helm values